### DISCUSSION TITLE
Joomla API / Web services future?

### DISCUSSION THREAD

**Discussion by n9iels** [joomla/joomla-cms#37908] [General]
Hi all,

I have not been around for a little while and not actively worked with Joomla! 4 yet, however I worked quite extensively with Joomla! 3. Last days I have been playing with the new integrated REST API, with the goal to see if I could use it to create a SPA for a Joomla! site. I have to say I am a bit... confused by the state of the project.

I was not able to find a lot of documentation except [this wiki page](https://docs.joomla.org/J4.x:Joomla_Core_APIs) with all the Core endpoints and [this page](https://docs.joomla.org/Joomla_Api_Specification) with some specifications. After fiddling for a while I got it working. The next goal was to limit permissions, so I created a new user group as direct child of "Public" and enable "Web services" login. This works, for the part to no articles can be added/modified/deleted via the API, however this API user can still see unpublished content or content that should only visible for registered users. Does this mean ACL is not completely implemented for the API? If so.. shouldn't there be a warning about this somewhere? I don't feel like this API is production ready at the moment.

Don't get me wrong here, I know Joomla! is open source and all developers are doing this in there own spare time (I contributed with a few PR's myself). So I am not demanding this to be fixed or something like that. I just can't stop wondering what the future vision is of the Joomla! Web Services API and why I cannot find anything about it. Is there anyone that can point me to the right direction? Anyone with other thoughts about this?

**Comment by alikon** [comment]
as a reference https://github.com/joomla/joomla-cms/issues/37925

**Comment by alikon** [comment]
maybe you could be interested on this one https://github.com/joomla/joomla-cms/pull/31579

**Comment by MarkRS-UK** [comment]
I'm just starting to use J4, I've been developing my own components in J3 for years, and I'm working heavily with the API.
I would be interested to cooperate in moving things forward with it, if indeed that's what's needed.

**Comment by TLWebdesign** [comment]
Should we try to get this some more attention? Could be a release feature for 5.2 for instance. Would be really neat to have more control on what an API user can do and can not do.

**Comment by webmasterab** [comment]
It would be very nice if it became possible to limit rights.

I notice that people are already regularly trying to do things with a token.
Fortunately, this is the wrong token.
Don't imagine that one day it will work and I will lose all the users or all the content.
